src/test/components/AppLocalizationProvider.test.js

```diff
@@ -9,51 +9,75 @@ import { AppLocalizationProvider } from 'firefox-profiler/components/app/AppLoca
 import { blankStore } from '../fixtures/stores';
 import { Provider } from 'react-redux';
 import {
-  getIsL10nFetching,
+  getl10nFetchingPhase,
   getLocalization,
 } from 'firefox-profiler/selectors/l10n';
 import { lazilyParsedBundles } from 'firefox-profiler/app-logic/l10n';
 import { requestL10n, receiveL10n } from 'firefox-profiler/actions/l10n';
-import { ReactLocalization } from '@fluent/react';
+import { ReactLocalization, Localized } from '@fluent/react';
+import { coerceMatchingShape } from '../../utils/flow';
+import { waitUntilState } from '../fixtures/utils';
 
 describe('AppLocalizationProvider', () => {
-  beforeEach(function() {
-    const response = (({
-      ok: true,
-      status: 200,
-      headers: {
-        get: () => 'text/plain',
-      },
-      text: () => Promise.resolve('resolved'),
-    }: any): Response);
-    window.fetch = jest.fn().mockResolvedValue(response);
-  });
+  function setupL10nFetch(language) {
+    jest.spyOn(window.navigator, 'languages', 'get').mockReturnValue(['en-US']);
+
+    const fetch = jest.fn().mockImplementation((fetchUrl: string) => {
+      if (fetchUrl === `/locales/${language}/app.ftl`) {
+        const response = (({
+          ok: true,
+          status: 200,
+          headers: {
+            get: () => 'text/plain',
+          },
+          text: () => Promise.resolve(`test-id = This is ${language} Text`),
```

Comment:
My suggestion was to using the `translatedText` here. That way there is going to be only one place to control/change this, which is good.